### ISSUE TITLE
undefined references to libm and pthread symbols when building Perl on top of compat/aarch64

### ISSUE THREAD

**Issue by boegel** [EESSI/software-layer#48]
When testing the installation of `Perl-5.30.2-GCCcore-9.3.0.eb`, I ran into linking errors in the `build` step:

```
opmini.o:opmini.c:function Perl_op_sibling_splice: error: undefined reference to 'pthread_getspecific'
...
toke.o:toke.c:function Perl_scan_num: error: undefined reference to 'pow'
```

For some reason, Perl's `Configure` script fails to find the necessary libraries to resolve these symbols:

```
Checking for optional libraries...
No -lcl.
No -lpthread.
No -lsocket.
No -linet.
No -lnsl.
No -lgdbm.
No -ldbm.
Found -ldb (shared).
No -lmalloc.
No -ldl.
No -lld.
No -lsun.
No -lm.
```

This was using the [Debian 10 container for build nodes](https://hub.docker.com/r/eessi/fuse-overlay/tags), on top of a compatibility layer built using the [CentOS 8 container for bootstrapping Prefix](https://hub.docker.com/r/eessi/bootstrap-prefix/tags).

In the `2020.10` revision of the EESSI pilot repo, this problem didn't pop up:

```
Checking for optional libraries...
No -lcl.
Found -lpthread (shared).
No -lsocket.
No -linet.
No -lnsl.
Found -lgdbm (shared).
No -ldbm.
Found -ldb (shared).
No -lmalloc.
Found -ldl (shared).
No -lld.
No -lsun.
Found -lm (shared).
```

In that case, the build node was running CentOS 8.2 (so not using the Debian 10 container for build nodes yet...).

So, somehow the OS of the build node is affecting how Perl's `Configure` script searches for libraries...

This problem didn't occur on `x86_64`, probably because of the difference in `lib` vs `lib64` directories in the Prefix installation (see https://github.com/EESSI/compatibility-layer/issues/63).

**Comment by boegel** [comment]
These issues are fixed with the updated to the Perl easyblock made in https://github.com/easybuilders/easybuild-easyblocks/pull/2268...

**Comment by boegel** [comment]
fixed `Perl` easyblock is included in EasyBuild v4.3.2, so closing this...
